src/persist-client/src/internal/paths.rs

```diff
@@ -124,6 +124,17 @@ impl PartialBatchKey {
         PartialBatchKey(format!("{}/{}", version, part_id))
     }
 
+    pub fn split(&self) -> (WriterKey, PartId) {
+        let (writer_key, part_id) = self
+            .0
+            .split_once('/')
+            .expect("partial batch key should contain a /");
+
+        let writer_key = WriterKey::from_str(writer_key).expect("correctly formatted writer key");
```

Comment:
won't this panic if it encounters a key written by an old version of persist?

also, this overlaps a bit with `BlobKey::parse_ids`, it'd be nice to dedup that if it's not too gnarly

Reply:
`WriterKey` has variants for both old-style and new-style keys, so this should parse successfully in both cases. (And the newer keys sort larger than all old-style keys, so the check below should also be correct.

Good call on the duplication - will see what I can do!